rules/REQUEST-932-APPLICATION-ATTACK-RCE.conf

```diff
@@ -359,6 +359,45 @@ SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAME
     setvar:'tx.inbound_anomaly_score_pl1=+%{tx.critical_anomaly_score}'"
 
 
+# [ Unix shell expressions - Bash Tilde expansion ]
+#
+# Detects the following patterns which are common in Unix shell scripts
+# and one-liners:
+#
+# ~+    $PWD
+# ~-    $OLDPWD
+# ~4    fifth directory from stack
+# ~-2   second directory from right to left
+#
+# Reference - https://linuxsimply.com/bash-scripting-tutorial/expansion/tilde-expansion/
+#
+# Regular expression generated from regex-assembly/932150.ra.
+# To update the regular expression run the following shell script
+# (consult https://coreruleset.org/docs/development/regex_assembly/ for details):
+#   crs-toolchain regex update 932150
+#
+SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|ARGS_NAMES|ARGS|XML:/* "@rx ~(?:\+|-[0-9]*|[0-9]+)" \
```

Comment:
Is there any merit in inspecting the URI / Filename or will this invariably lead to errors? There is this old convention of hosting user webpages under `example.com/~john`

Reply:
An alphabet after the tilde(~) is not valid expansion syntax, so REQUEST_FILENAME won't make much sense, unless if the pattern is `/~3john` or something similar.

Reply:
The question is more if you can do this type of injection in the URI.

But given all the other 932 rules are not working on the URI, this question is of no concern here.

Reply:
You're right, I haven't thought about the URI, but as you said 932 rules exclude it, so it's no concern here.